pkg/kube/krt/collection.go

```diff
@@ -46,24 +44,21 @@ type dependencyState[I any] struct {
 	// Stores a map of I -> secondary dependencies (added via Fetch)
 	objectDependencies  map[Key[I]][]*dependency
 	indexedDependencies map[indexedDependency]sets.Set[Key[I]]
-	// indexedDependenciesExtractor stores a map of [collection,fetch type] => an extractor to get change keys.
-	// Note that a given collection can have multiple Fetches, but they are limited to those of the different kind.
-	// I.e. you can do a `Fetch(c1, FilterIndex()) + Fetch(c1, FilterKey())` but not `Fetch(c1, FilterIndex(idxA)) + Fetch(c1, FilterIndex(idxB))`.
-	// Multiple `FetchIndex` within a single transformation must use the same index.
-	// This only applies within a single transformation; it is fine to fetch the the same `c1` in any way from different collections.
+	// indexedDependenciesExtractor stores a map of [collection,fetch type,(optional)index id] => an extractor to get change keys.
```

Comment:
If I'm reading this correctly there is still a limitation for "key" type filters.  We would support `0-1 filter keys + 0-N filter indexes`.  Can we note the new state rather than deleting the comment on limitations entirely?

Reply:
Fair. I was thinking about doing something like this to also allow multiple fetches of any kind.

```go
	if f.keys.Len() > 0 {
		if f.index != nil {
			panic("cannot filter by index and key")
		}
                keysFilterHash := uint64(0)
                for _, key := f.Keys.List() {
                  // build up a hash
                }
		return f.keys.List(), getKeyType, getKeyExtractor, keysFilterHash, true
	}

```

Reply:
That works, if the intent is to handle keys as well then we can ignore "fixing" the comment about limitations.

Reply:
just pushed a test that I think confirms that was safe to begin with